arclight/app/models/concerns/arclight/catalog.rb

```diff
@@ -0,0 +1,47 @@
+# frozen_string_literal: true
+
+module Arclight
+  ##
+  # Arclight specific methods for the Catalog
+  module Catalog
+    extend ActiveSupport::Concern
+
+    included do
+      before_action only: :index do
+        if (params.dig(:f, :collection) || []).any?(&:blank?)
+          params[:f][:collection].compact_blank!
+          params[:f].delete(:collection) if params[:f][:collection].blank?
+        end
+      end
+
+      before_action only: :hierarchy do
+        blacklight_config.search_state_fields += %i[id limit offset]
+      end
+
+      Blacklight::Configuration.define_field_access :summary_field, Blacklight::Configuration::ShowField
+      Blacklight::Configuration.define_field_access :background_field, Blacklight::Configuration::ShowField
+      Blacklight::Configuration.define_field_access :related_field, Blacklight::Configuration::ShowField
+      Blacklight::Configuration.define_field_access :indexed_terms_field, Blacklight::Configuration::ShowField
+      Blacklight::Configuration.define_field_access :in_person_field, Blacklight::Configuration::ShowField
+      Blacklight::Configuration.define_field_access :cite_field, Blacklight::Configuration::ShowField
+      Blacklight::Configuration.define_field_access :contact_field, Blacklight::Configuration::ShowField
+      Blacklight::Configuration.define_field_access :component_field, Blacklight::Configuration::ShowField
+      Blacklight::Configuration.define_field_access :component_indexed_terms_field, Blacklight::Configuration::ShowField
+      Blacklight::Configuration.define_field_access :terms_field, Blacklight::Configuration::ShowField
+      Blacklight::Configuration.define_field_access :component_terms_field, Blacklight::Configuration::ShowField
+      Blacklight::Configuration.define_field_access :group_header_field, Blacklight::Configuration::IndexField
+    end
+
+    def hierarchy
+      # Remove the fq (preview=false) parameter from the hierarchy call
+      # this ensures that that collection contents can be viewed for previewed items
+      blacklight_config.default_solr_params.delete(:fq)
```

Comment:
I wonder if this could also be done by using the [SearchBuilder processor chain](https://github.com/projectblacklight/blacklight/wiki/Extending-or-Modifying-Blacklight-Search-Behavior#extending-the-searchbuilder) as that is already an [extension point in our arclight app](https://github.com/ucldc/cinco/blob/main/arclight/app/models/search_builder.rb)